src/tools/wasm-fuzz-lattices.cpp

```diff
@@ -235,6 +250,14 @@ RandomFullLattice::Element RandomFullLattice::makeElement() const noexcept {
     return ElementImpl{typename ArrayFullLattice::Element{
       l->lattice.makeElement(), l->lattice.makeElement()}};
   }
+  if (const auto* l = std::get_if<Vector<RandomFullLattice>>(lattice.get())) {
+    std::vector<typename RandomFullLattice::Element> elem;
+    elem.reserve(l->size);
+    for (size_t i = 0; i < l->size; ++i) {
+      elem.push_back(l->lattice.makeElement());
+    }
+    return ElementImpl{std::move(elem)};
+  }
   WASM_UNREACHABLE("unexpected lattice");
```

Comment:
I won't ask for a change here, but I really do think `virtual` would be a nicer way to write code like this, rather than have to keep modifying RandomFullLattice for each type. That is, I would personally gone the other route and had a virtual method that this class calls, each new lattice just implements a "makeRandom" interface, so it all plugs in.

Reply:
I plan to at least get rid of the distinction between `RandomFullLattice` and `RandomLattice`, which should simplify things.

I can also clean up this giant if-else chain by using `std::visit` combined with overloaded free functions, which will be closer to what you're envisioning, I think. I don't think that going all the way to creating an inheritance-based wrapper for each of the lattices would be worth it, though.